metadata-service/restli-servlet-impl/src/main/java/com/linkedin/metadata/resources/entity/AspectResource.java

```diff
@@ -235,56 +238,87 @@ public Task<String> ingestProposal(
+
+    // Ingest Authorization Checks
+    List<Pair<MetadataChangeProposal, Integer>> exceptions = isAPIAuthorized(authentication, _authorizer, ENTITY,
+             opContext.getEntityRegistry(), metadataChangeProposals)
              .stream().filter(p -> p.getSecond() != HttpStatus.S_200_OK.getCode())
              .collect(Collectors.toList());
-     if (!exceptions.isEmpty()) {
-         throw new RestLiServiceException(
-                 HttpStatus.S_403_FORBIDDEN, "User is unauthorized to modify entity: " + exceptions.stream()
+    if (!exceptions.isEmpty()) {
+        String errorMessages = exceptions.stream()
                  .map(ex -> String.format("HttpStatus: %s Urn: %s", ex.getSecond(), ex.getFirst().getEntityUrn()))
-                 .collect(Collectors.toList()));
-     }
-
+                 .collect(Collectors.joining(", "));
+        throw new RestLiServiceException(
+                 HttpStatus.S_403_FORBIDDEN, "User is unauthorized to modify entity: " + errorMessages);
+    }
     String actorUrnStr = authentication.getActor().toUrnStr();
     final AuditStamp auditStamp =
         new AuditStamp().setTime(_clock.millis()).setActor(Urn.createFromString(actorUrnStr));
 
     return RestliUtil.toTask(() -> {
-      log.debug("Proposal: {}", metadataChangeProposal);
+      log.debug("Proposals: {}", metadataChangeProposals);
       try {
         final AspectsBatch batch = AspectsBatchImpl.builder()
-                .mcps(List.of(metadataChangeProposal), auditStamp, opContext.getRetrieverContext().get())
+                .mcps(metadataChangeProposals, auditStamp, opContext.getRetrieverContext().get())
                 .build();
 
         Set<IngestResult> results =
                 _entityService.ingestProposal(opContext, batch, asyncBool);
 
-            java.util.Optional<IngestResult> one = results.stream().findFirst();
+            for (IngestResult result : results) {
+                // Update runIds, only works for existing documents, so ES document must exist
+                Urn resultUrn = result.getUrn();
 
-            // Update runIds, only works for existing documents, so ES document must exist
-            Urn resultUrn = one.map(IngestResult::getUrn).orElse(metadataChangeProposal.getEntityUrn());
-            if (one.map(result -> result.isProcessedMCL() || result.isUpdate()).orElse(false)) {
-              tryIndexRunId(opContext,
-                  resultUrn, metadataChangeProposal.getSystemMetadata(), entitySearchService);
+                if (resultUrn != null && (result.isProcessedMCL() || result.isUpdate())) {
+                    tryIndexRunId(opContext, resultUrn, result.getRequest().getSystemMetadata(), entitySearchService);
+                }
             }
-            return resultUrn.toString();
+
+            // TODO: We don't actually use this return value anywhere. Maybe we should just stop returning it altogether?
+            return "success";
```

Comment:
I do think it would be better to return the list of urns that you successfully ingested in the response here to keep in line with what was happening previously, but it is unlikely to cause issues. Just returning "success" does not provide any information beyond what the status code already provides.

Reply:
imo returning the urn also doesn't provide much information either - only the status code is important. Ingestion also only looks at the status code, and not the body

I did this for simplicity - I wanted to keep the return type `Task<String>` for both methods so that the code would be less complex. I can refactor it if you think it's important, but imo we don't gain anything by doing it